posthog/src/main/java/com/posthog/PostHogIntegration.kt

```diff
@@ -0,0 +1,17 @@
+package com.posthog
+
+/**
+ * Integration interface for capturing events automatically or adding plugins to the PostHog SDK
+ */
+public interface PostHogIntegration {
```

Comment:
this is cool... can an SDK user add integrations too?

Should we have hooks in this interface that an integration has to consider?

Reply:
Yep, they can add integrations as well that get automatically installed/uninstalled during the SDK's lifecycle.
All the events that are automatically captured are already integrations, its a plugin system.

Reply:
That's really cool. Playing the role of stupid user - how do I find out more?

Reply:
Right now there are no docs since it's a new approach, once we get closer to a stable version, I will add a new section to https://posthog.com/docs/libraries/android but would like to focus on the existing features for now.